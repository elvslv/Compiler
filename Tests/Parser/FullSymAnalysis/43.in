var
  n, m, l, r, i, t, j: integer;
  a: array[1..1000] of integer;
  len: array[1..2000] of integer;
  p, q: array[1..1000] of integer;
  ans: integer;
begin  
for i := l to r do
  begin
    repeat
     ans := ans + a[q[l]];
     if j = 7 then
	break
     else
       continue;
    until l < r;	
    t := m + n;
    j := i;
    while t > n do
    begin
      if j <= len[p[t - n]] then
        t := p[t - n]
      else
      begin
        j := j - len[p[t - n]];
        t := q[t - n];
      end;
    end;
    ans := ans + a[t];
  end;
end.