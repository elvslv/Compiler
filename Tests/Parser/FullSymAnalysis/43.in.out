var A : array [const 1..const 1000] of INTEGER;
var ANS : INTEGER;
var I : INTEGER;
var J : INTEGER;
var L : INTEGER;
var LEN : array [const 1..const 2000] of INTEGER;
var M : INTEGER;
var N : INTEGER;
var P : array [const 1..const 1000] of INTEGER;
var Q : array [const 1..const 1000] of INTEGER;
var R : INTEGER;
var T : INTEGER;

begin                                            
|                                                
|___for                                          
|   |                                            
|   |___:=                                       
|   |   |                                        
|   |   |___I                                    
|   |   |                                        
|   |   |___L                                    
|   |                                            
|   to                                           
|   |                                            
|   |___R                                        
|   |                                            
|   do                                           
|   |                                            
|   |___begin                                    
|       |                                        
|       |___repeat                               
|       |   |                                    
|       |   |___begin                            
|       |   |   |                                
|       |   |   |___:=                           
|       |   |   |   |                            
|       |   |   |   |___ANS                      
|       |   |   |   |                            
|       |   |   |   |___+                        
|       |   |   |       |                        
|       |   |   |       |___ANS                  
|       |   |   |       |                        
|       |   |   |       |___[]                   
|       |   |   |           |                    
|       |   |   |           |___A                
|       |   |   |           |                    
|       |   |   |           |___[]               
|       |   |   |               |                
|       |   |   |               |___Q            
|       |   |   |               |                
|       |   |   |               |___L            
|       |   |   |                                
|       |   |   |___if                           
|       |   |   |   |                            
|       |   |   |   |___=                        
|       |   |   |   |   |                        
|       |   |   |   |   |___J                    
|       |   |   |   |   |                        
|       |   |   |   |   |___7                    
|       |   |   |   |                            
|       |   |   |   then                         
|       |   |   |   |                            
|       |   |   |   |___begin                    
|       |   |   |   |   |                        
|       |   |   |   |   |___break                
|       |   |   |   |   |                        
|       |   |   |   |   end                      
|       |   |   |   |                            
|       |   |   |   else                         
|       |   |   |   |                            
|       |   |   |   |___begin                    
|       |   |   |       |                        
|       |   |   |       |___continue             
|       |   |   |       |                        
|       |   |   |       end                      
|       |   |   |                                
|       |   |   end                              
|       |   |                                    
|       |   until                                
|       |   |                                    
|       |   |___<                                
|       |       |                                
|       |       |___L                            
|       |       |                                
|       |       |___R                            
|       |                                        
|       |___:=                                   
|       |   |                                    
|       |   |___T                                
|       |   |                                    
|       |   |___+                                
|       |       |                                
|       |       |___M                            
|       |       |                                
|       |       |___N                            
|       |                                        
|       |___:=                                   
|       |   |                                    
|       |   |___J                                
|       |   |                                    
|       |   |___I                                
|       |                                        
|       |___while                                
|       |   |                                    
|       |   |___>                                
|       |   |   |                                
|       |   |   |___T                            
|       |   |   |                                
|       |   |   |___N                            
|       |   |                                    
|       |   do                                   
|       |   |                                    
|       |   |___begin                            
|       |       |                                
|       |       |___if                           
|       |       |   |                            
|       |       |   |___<=                       
|       |       |   |   |                        
|       |       |   |   |___J                    
|       |       |   |   |                        
|       |       |   |   |___[]                   
|       |       |   |       |                    
|       |       |   |       |___LEN              
|       |       |   |       |                    
|       |       |   |       |___[]               
|       |       |   |           |                
|       |       |   |           |___P            
|       |       |   |           |                
|       |       |   |           |___-            
|       |       |   |               |            
|       |       |   |               |___T        
|       |       |   |               |            
|       |       |   |               |___N        
|       |       |   |                            
|       |       |   then                         
|       |       |   |                            
|       |       |   |___begin                    
|       |       |   |   |                        
|       |       |   |   |___:=                   
|       |       |   |   |   |                    
|       |       |   |   |   |___T                
|       |       |   |   |   |                    
|       |       |   |   |   |___[]               
|       |       |   |   |       |                
|       |       |   |   |       |___P            
|       |       |   |   |       |                
|       |       |   |   |       |___-            
|       |       |   |   |           |            
|       |       |   |   |           |___T        
|       |       |   |   |           |            
|       |       |   |   |           |___N        
|       |       |   |   |                        
|       |       |   |   end                      
|       |       |   |                            
|       |       |   else                         
|       |       |   |                            
|       |       |   |___begin                    
|       |       |       |                        
|       |       |       |___:=                   
|       |       |       |   |                    
|       |       |       |   |___J                
|       |       |       |   |                    
|       |       |       |   |___-                
|       |       |       |       |                
|       |       |       |       |___J            
|       |       |       |       |                
|       |       |       |       |___[]           
|       |       |       |           |            
|       |       |       |           |___LEN      
|       |       |       |           |            
|       |       |       |           |___[]       
|       |       |       |               |        
|       |       |       |               |___P    
|       |       |       |               |        
|       |       |       |               |___-    
|       |       |       |                   |    
|       |       |       |                   |___T
|       |       |       |                   |    
|       |       |       |                   |___N
|       |       |       |                        
|       |       |       |___:=                   
|       |       |       |   |                    
|       |       |       |   |___T                
|       |       |       |   |                    
|       |       |       |   |___[]               
|       |       |       |       |                
|       |       |       |       |___Q            
|       |       |       |       |                
|       |       |       |       |___-            
|       |       |       |           |            
|       |       |       |           |___T        
|       |       |       |           |            
|       |       |       |           |___N        
|       |       |       |                        
|       |       |       end                      
|       |       |                                
|       |       end                              
|       |                                        
|       |___:=                                   
|       |   |                                    
|       |   |___ANS                              
|       |   |                                    
|       |   |___+                                
|       |       |                                
|       |       |___ANS                          
|       |       |                                
|       |       |___[]                           
|       |           |                            
|       |           |___A                        
|       |           |                            
|       |           |___T                        
|       |                                        
|       end                                      
|                                                
end                                              
